src/query/block/types.go

```diff
@@ -75,12 +104,31 @@ type StepIter interface {
 	Meta() Metadata
 }
 
+// UnconsolidatedStepIter iterates through a block vertically
+type UnconsolidatedStepIter interface {
+	Iterator
+	// Current returns the current step for the block
+	Current() (UnconsolidatedStep, error)
+	// StepCount returns the number of steps
+	StepCount() int
+	// SeriesMeta returns the metadata for each series in the block
+	SeriesMeta() []SeriesMeta
+	// Meta returns the metadata for the block
+	Meta() Metadata
```

Comment:
Can roll these into a single interface and share that with `StepIter`

Reply:
Actually, might be a good idea to pull 
```
		// SeriesMeta returns the metadata for each series in the block
	SeriesMeta() []SeriesMeta
	// Meta returns the metadata for the block
	Meta() Metadata
```
into a single interface since a lot of these use them